pkg/morph/client/notifications.go

```diff
@@ -120,22 +120,23 @@ func (c *Client) ReceiveNotaryRequests(txSigner util.Uint160) error {
 		return ErrConnectionLost
 	}
 
-	c.subs.Lock()
-	defer c.subs.Unlock()
+	c.subs.RLock()
 
-	if c.subs.subscribedToAllNotaryEvents {
+	if _, ok := c.subs.subscribedNotaryEvents[txSigner]; ok || c.subs.subscribedToAllNotaryEvents {
+		c.subs.RUnlock()
 		return nil
 	}
 
-	if _, ok := c.subs.subscribedNotaryEvents[txSigner]; ok {
-		return nil
-	}
+	c.subs.RUnlock()
 
 	_, err := c.client.ReceiveNotaryRequests(&neorpc.TxFilter{Signer: &txSigner}, c.subs.curNotaryChan)
```

Comment:
Can we have two threads here?

Reply:
do you mean subscribing? two routines that subscribe for smth?